apps/src/music/views/MusicPlayView.tsx

```diff
@@ -30,76 +42,107 @@ const MusicPlayView: React.FunctionComponent<MusicPlayViewProps> = ({
     [projectName]
   );
 
-  // Share button will only appear when users browser supports the Web Share API
+  // Share button will only appear when user's browser supports the Web Share API.
   // (Can be mobile browsers and some desktop browser like macOS Safari)
-  // Requires HTTPS connection (adhoc or production page).
-  // Was unable to access it on localhost with our routing (http://localhost-studio.code.org:3000/
-  // is not considered a safe URL by browser, only http://localhost:3000/ is).
+  // Requires HTTPS connection.
+  // For testing purposes, we can pass a query parameter canShare=true to force the button to appear.
+  const canShareInternal = queryParams('canShare') === 'true';
   const canShare =
-    navigator && navigator.canShare && navigator.canShare(shareData);
+    (navigator && navigator.canShare && navigator.canShare(shareData)) ||
+    canShareInternal;
 
+  const projectManager = Lab2Registry.getInstance().getProjectManager();
   const onShareProject = useCallback(() => {
     navigator?.share(shareData);
   }, [shareData]);
   const onViewCode = useCallback(() => {
-    console.log('view code');
-  }, []);
+    if (projectManager) {
+      projectManager.flushSave().then(() => {
+        projectManager.redirectToView();
+      });
+    }
+  }, [projectManager]);
   const onRemix = useCallback(() => {
-    console.log('make my own');
-  }, []);
+    if (projectManager) {
+      projectManager.flushSave().then(() => {
+        projectManager.redirectToRemix();
+      });
+    }
+  }, [projectManager]);
 
   return (
     <div className={moduleStyles.musicPlayViewContainer}>
-      <div className={moduleStyles.musicPlayViewCard}>
-        <div className={moduleStyles.musicPlayViewInfoSection}>
-          <img src={codeOrgLogo} alt="Code.org" />
-          <Heading2>{projectName}</Heading2>
-          <BodyTwoText>Built with Music Lab</BodyTwoText>
-        </div>
-
-        <div className={moduleStyles.musicPlayViewPlaySection}>
-          <Button
-            isIconOnly={true}
-            icon={{iconStyle: 'solid', iconName: !isPlaying ? 'play' : 'stop'}}
-            onClick={!isPlaying ? onPlay : onStop}
-            size="s"
-            color="white"
-            type="secondary"
-          />
-          {/*TODO: get the total length of the song, show current player position/play progress*/}
-          <input type="range" min="0" max="100" />
-        </div>
+      <div className={moduleStyles.musicPlayViewCardContainer}>
+        <img
+          className={moduleStyles.musicPlayViewImage}
+          src={musicPlayViewLogo}
+          alt="Code.org play view logo"
+        />
+        <div className={moduleStyles.musicPlayViewCard}>
+          <div className={moduleStyles.musicPlayViewInfoSection}>
+            <Heading2 className={moduleStyles.musicPlayViewInfoText}>
```

Comment:
nit: (just a comment, no need to update this) Coming back to the conversation re setting a className to each specific text element, now that we have a live example, in my opinion it's making code harder to read and adding extra complexity. Personal preference, but I like more when all the general styles are stored in some-level-container and classNames applied to text only when text is styled in 'exceptional' way (like emphasised text, error message, etc)

cc @sanchitmalhotra126

Reply:
@sanchitmalhotra126 - lmk what you think. Thanks y'all!

Reply:
Hmm, I think I personally prefer the classnames still since it's easier to look up the styles in the stylesheet? That being said, might be good to make these names less verbose if it makes it easier to read by removing the `musicPlayView` prefix (e.g. just `card`, `infoSection`, etc). The class names are already scoped to the specific module and get converted to IDs in production, so we don't need to worry about naming collisions.

Reply:
Thanks! Will shorten style names.